src/iiif/views.py

```diff
@@ -1,28 +1,50 @@
 import logging
+from datetime import datetime
 
+import jwt
 from django.conf import settings
 from django.http import HttpResponse
 from django.views.decorators.csrf import csrf_exempt
+from jwt.exceptions import ExpiredSignatureError, InvalidSignatureError
 from requests.exceptions import RequestException
 
-from . import tools
+from iiif import tools
 
 log = logging.getLogger(__name__)
 
 RESPONSE_CONTENT_NO_TOKEN = "No token supplied"
+RESPONSE_CONTENT_JWT_TOKEN_EXPIRED = "Your token has expired. Request a new token."
 RESPONSE_CONTENT_NO_DOCUMENT_IN_METADATA = "Document not found in metadata"
 RESPONSE_CONTENT_ERROR_RESPONSE_FROM_METADATA_SERVER = "The iiif-metadata-server cannot be reached"
 RESPONSE_CONTENT_ERROR_RESPONSE_FROM_CANTALOUPE = "The iiif-image-server cannot be reached"
 
 
 @csrf_exempt
 def index(request, iiif_url):
-    if not settings.DATAPUNT_AUTHZ['ALWAYS_OK']:
-        if not request.META.get('HTTP_AUTHORIZATION', None):
-            return HttpResponse(RESPONSE_CONTENT_NO_TOKEN, status=401)
-        token = request.META['HTTP_AUTHORIZATION']
+    if not request.META.get('HTTP_AUTHORIZATION') and not request.GET.get('auth'):
+        return HttpResponse(RESPONSE_CONTENT_NO_TOKEN, status=401)
+
+    # Check the jwt token if necessary
```

Comment:
Je gaat nu zelf json web tokens checken, ik twijfel of dat wenselijk is. Mijn eerste gevoel zegt dat het beter is daar bestaande oplossingen voor te gebruiken.